packages/core/src/components/carousel/carousel.tsx

```diff
@@ -1,142 +1,201 @@
-
-    const children = this.host.querySelectorAll('atom-carousel-item')
-
-    children.forEach((child) => {
-      const slide = child.querySelector('swiper-slide')
-
-      if (slide) {
-        this.swiperEl.appendChild(slide)
-        this.host.removeChild(child)
-      }
-    })
-
-    this.swiperEl.swiper?.on('slideChange', () => {
-      this.atomChange.emit()
-    })
-    this.swiperEl.swiper?.on('navigationPrev', () => {
-      this.atomClickPrev.emit()
-    })
-    this.swiperEl.swiper?.on('navigationNext', () => {
-      this.atomClickNext.emit()
-    })
-
-    let params: { pagination: PaginationOptions } = this.pagination && {
-      pagination: {
-        type:
-          this.paginationType === 'thumbnails' ? 'custom' : this.paginationType,
-        clickable: this.paginationClickable,
-      },
+    this.carouselWrapper =
+      this.element.shadowRoot.querySelector('.carousel-wrapper')
+    this.nextButton = this.element.shadowRoot.querySelector('.navigation--next')
+    this.prevButton = this.element.shadowRoot.querySelector('.navigation--prev')
+
+    this.carouselWrapper.addEventListener('touchstart', this.handleTouchStart)
+    this.carouselWrapper.addEventListener('touchend', this.handleTouchEnd)
+
+    this.showOrHideNavigationButtons()
+
+    if (this.autoplay > 0) this.startAutoplay()
+  }
+
+  disconnectedCallback() {
+    this.carouselWrapper.removeEventListener(
+      'touchstart',
+      this.handleTouchStart
+    )
+    this.carouselWrapper.removeEventListener('touchend', this.handleTouchEnd)
+  }
+
+  handleMoveToPaginationItem(index: number) {
+    if (!this.carouselItems || !this.carouselItems[index]) return
+
+    this.currentIndex = index
+    this.showOrHideNavigationButtons()
+  }
+
+  handleNavigationClick(event: Event) {
+    const button = event.currentTarget as HTMLButtonElement
+    const isNext = button?.classList.contains('navigation--next')
+    let newIndex = this.currentIndex + (isNext ? 1 : -1)
+
+    const isLastItemAndNext =
+      isNext && this.currentIndex === this.carouselItems.length - 1
+    const isFirstItemAndPrev = !isNext && this.currentIndex === 0
+
+    if (!this.loop && (isLastItemAndNext || isFirstItemAndPrev)) return
+
+    if (this.loop) {
+      newIndex =
+        (newIndex + this.carouselItems.length) % this.carouselItems.length
     }
 
-    if (this.paginationType === 'thumbnails') {
-      const urls = this.thumbnailImages ? this.thumbnailImages.split(',') : []
+    this.currentIndex = newIndex
+    this.showOrHideNavigationButtons()
+  }
+
+  showOrHideNavigationButtons() {
+    if (!this.nextButton || !this.prevButton) return
 
-      params = {
-        pagination: {
-          ...params.pagination,
-          renderCustom: (_swiper, current, total) =>
-            renderThumbs(current, total, urls, this.videoIcons),
-        },
-      }
+    if (this.loop) {
+      this.nextButton.classList.remove('disabled')
+      this.prevButton.classList.remove('disabled')
```

Comment:
If `loop` is true, buttons always be visible. Why remove here instead of `componentDidLoad` ? I think this way, you won't need pass for this with every click

Reply:
Perfect, we don't need to check it, done [fe89528](https://github.com/juntossomosmais/atomium/pull/434/commits/fe89528fdb8c8053931afef27f8e9a9c0f78ba07)